test/mlir/affine_conversion/core_ops.mlir

```diff
@@ -18,6 +18,108 @@ func @simple_gather(%arg0: !ng.tensor<16x!ng.i64>, %arg1: !ng.tensor<512x32xf32>
 
 // -----
 
+// Equal Op
+// CHECK:       affine.for %[[I:.*]] = 0 to 2
+// CHECK-NEXT:  affine.for %[[J:.*]] = 0 to 2
+// CHECK-NEXT:  %[[C1:.*]] = constant
+// CHECK-NEXT: 	%[[C2:.*]] = constant
+// CHECK:	%[[O1:.*]] = affine.load  %{{.*}}[%[[I]], %[[J]]]
+// CHECK:	%[[O2:.*]] = affine.load  %{{.*}}[%[[I]], %[[J]]]
+// CHECK-NEXT:	%[[R1:.*]] = cmpf "oeq"
+// CHECK:       %[[R2:.*]] = select
+// CHECK-NEXT:  affine.store %[[R2]], %{{.*}}[%[[I]], %[[J]]]
```

Comment:
They look great! Thanks! Some suggestions:
- I think we should verify that the `cmpf` operand are actually `O1` and `O2` and that they are in the right order.
- Same for select (and therefore, constant ops): we should check operands to make sure the order is correct.
- We can add a `CHECK-LABEL: func @simple_equal` to make sure we prevent other checks from matching against other functions.
- nit: it would be great if you could use the same indentation for all the instructions within the same loop. Otherwise, they look like they are at different nesting levels.

Same comments for tests below.